src/extension/features/budget/subtract-upcoming-from-available/index.js

```diff
@@ -0,0 +1,124 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { isCurrentRouteBudgetPage } from 'toolkit/extension/utils/ynab';
+import { getEmberView } from 'toolkit/extension/utils/ember';
+import { formatCurrency } from 'toolkit/extension/utils/currency';
+import { addToolkitEmberHook } from 'toolkit/extension/utils/toolkit';
+
+let totalUpcoming = 0;
+
+export class SubtractUpcomingFromAvailable extends Feature {
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage();
+  }
+
+  invoke() {
+    addToolkitEmberHook(this, 'budget-bar-v2', 'didRender', this.run);
+    addToolkitEmberHook(this, 'budget-bar-v2', 'didUpdate', this.run);
+
+    addToolkitEmberHook(this, 'budget/inspector/default-inspector', 'didRender', this.run);
+    addToolkitEmberHook(this, 'budget/inspector/multi-select-inspector', 'didRender', this.run);
+  }
+
+  run() {
+    if (!this.shouldInvoke()) return;
+    this.updateAvailableBalance();
+    this.addTotalAvailableAfterUpcoming();
+  }
+
+  updateAvailableBalance() {
+    totalUpcoming = 0;
+
+    $('.budget-table-row.is-sub-category').each((_, element) => {
```

Comment:
Once you update `addToolkitEmberHook` to use `budget-table-row`, the function you provide (`this.run`) will be called with the row element itself that changed and getting all rows won't be necessary.

you should be able to basically remove this each, add `element` as an argument to `updateAvailableBalance` and be good to go! (+/- a few other changes you may have to sort out :P)